.github/ISSUE_TEMPLATE/feature_request_form.yml

```diff
@@ -0,0 +1,107 @@
+      options:
+        - New Feature
+        - Improvement
+        - Change
+    validations:
+      required: true
+
+  - type: dropdown
+    id: criticality
+    attributes:
+      label: How would you describe the priority of this feature request
+      options:
+        - Critical (currently preventing usage)
+        - High
+        - Medium
+        - Low (would be nice)
+    validations:
+      required: true
+
+  - type: textarea
+    id: problem
+    attributes:
+      label: Please provide a clear description of problem this feature solves
+      description: Real usage examples are especially helpful, non-code.
+    validations:
+      required: true
+
+  - type: textarea
+    id: Feature_Description
+    attributes:
+      label: Feature Description
+      description: Please provide clear description of the feature you request (refer to [User Story format](https://www.atlassian.com/agile/project-management/user-stories#:~:text=User%20story%20template%20and%20examples) and [EARS format](https://ieeexplore.ieee.org/document/5328509))
+      placeholder: >
+        For new feature request, please use one of the following format to describe the feature
+          1. From End-user perspective, use the following user story format
+              As a <persona>, I <want to>, <so that>.
+          2. From System perspective, use the following EARS format
+              <Pre-Condition> <System> shall  <process> <object to be process> <condition>
+        For changing or improving existing feature, it's recommended to provide the previous Feature Request ID.
+    validations:
+      required: true
+
+  - type: textarea
+    id: solution
+    attributes:
+      label: Describe your ideal solution
+      description: Please describe the functionality you would like added.
+      placeholder: >
+        A new function that takes in the information in this form, and triages the issue
+
+        def feature_request(form_info):
+            parse(form_info)
+            return triage_outcome
+        You can even provide a reference implementation in other library.
+    validations:
+      required: true
+
+  - type: textarea
+    id: alternatives
+    attributes:
+      label: Describe any alternatives you have considered
+      description: List any other libraries, or approaches you have looked at or tried.
+      placeholder: I have looked at library xyz and qrs, but they do not offer GPU acceleration
+
+  - type: textarea
+    id: misc
+    attributes:
+      label: Additional context
+      description: Add any other context, code examples, or references to existing implementations about the feature request here. If applicable, please list the modules affected.
+
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Check for duplicates
+      options:
+        - label: I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report
+          required: true
```

Comment:
Github points out the lack of newline at the end of the file. I don't know if we want it or not, just wanted to note this.